plugins/osdn/vnids.go

```diff
@@ -18,14 +20,103 @@ const (
 	AdminVNID = uint(0)
 )
 
+func (oc *OsdnController) GetVNID(name string) (uint, error) {
+	oc.vnidLock.Lock()
+	defer oc.vnidLock.Unlock()
+
+	if id, ok := oc.vnidMap[name]; ok {
+		return id, nil
+	}
+	// In case of error, return some value which is not a valid VNID
+	return MaxVNID + 1, fmt.Errorf("Failed to find NetID for namespace: %s in vnid map", name)
+}
+
+// Nodes asynchronously watch for both NetNamespaces and services
+// NetNamespaces populates vnid map and services/pod-setup depend on vnid map
+// If for some reason, vnid map propagation from master to node is slow
+// and if service/pod-setup tries to lookup vnid map then it may fail.
+// So, use this method to alleviate this problem. This method will
+// retry vnid lookup and NetNamespaces in etcd before giving up.
+func (oc *OsdnController) TryAndGetVNID(name string) (uint, error) {
```

Comment:
"try" suggests something like `pthread_trylock` to me, which is exactly the opposite of what it actually means here. Maybe `WaitForVNID`?

Also, some of the changes here should be pushed back into the previous commit; it's weird to add a "retry" flag in the previous commit and then remove it in this one and do things differently instead.